File: clinic/clinic/report/appointment_analytics/appointment_analytics.py
import frappe 
import json
from frappe.utils import getdate
from frappe.utils import add_to_date


@frappe.whitelist()
def sql_call(from_date,to_date,args):
    """
    from_date,to_date: is date data type
    args is an dict of dict that has only two values doctor and status
    this function returns the specific doctor schedule
    """
    args= json.loads(args)
    from_date = getdate(from_date)
    to_date = getdate(to_date)
    result = []

    sql = """
            select 
            CANAME as 'name',
            Clinic AS 'clinic',
            Doctor AS 'doctor',
            customer as customer,
            Status as 'status',
            T.Billed as 'Billed:Int:100',
            T.partially as 'partially Billed:Int:100'
            from
            (select 
            PA.name as 'CANAME',
            PA.patient_name as 'customer',
            PA.clinic as 'Clinic',
            PA.doctor_name as 'Doctor',
            PA.status as 'Status',
            (CASE WHEN SI.status = 'Paid' THEN 1 ELSE 0 END) as 'Billed',
            (CASE WHEN SI.status = 'Overdue' OR SI.status= 'Partly Paid' THEN 1 ELSE 0 END) as 'partially'
            FROM `tabClient Appointment CT` AS PA
            LEFT JOIN `tabSales Invoice` SI
            on PA.NAME = SI.appointment
            WHERE PA.appointment_date >= %(from_date)s 
            and PA.appointment_date <= %(to_date)s 
            and PA.doctor_name = %(doctor)s 
            and PA.status = %(status)s
            and PA.status != 'Cancelled') as T
            order by 
            T.clinic,
            T.Doctor,
            T.status





    """
    for item in args.values():
        data = {"from_date":from_date,"to_date":to_date,"doctor":item['doctor'],'status':item['status']}
        result= result + frappe.db.sql(sql,data,as_dict=1)
       
        pass
    
       
    return result
    pass

File: clinic/clinic/report/appointment_analytics/appointment_analytics.js

var x = {};

frappe.query_reports["Appointment Analytics"] = {
	"filters": [

		{
			"fieldname":"from_date",
			"label": __("From Date"),
			"fieldtype": "Date",
			"width": "80",
			"default": frappe.datetime.get_today()
		},
		{
			"fieldname":"to_date",
			"label": __("To Date"),
			"fieldtype": "Date",
			"width": "80",
			"default": frappe.datetime.get_today()
		}
	],
	"get_datatable_options":function(options) {

		return Object.assign(options,{
			checkboxColumn: true,
			events:{
				onCheckRow: function (data) {
					if(!data)
						return;
					let index = data[1].content
					if(!x[index])
					{
						x[index] = {'doctor':data[3].content,
									'status':data[4].content
						}
					}
					else{
						delete(x[index])
					}
					let from_date = frappe.query_report.filters[0].value;
					let to_date = frappe.query_report.filters[1].value;
					
					console.log(x)
			
					
				},
				
			}
		})
		},
		"onload":function(report){
			report.page.add_inner_button(__('more'), function () {
				frappe.call({
					method: "clinic.clinic.report.appointment_analytics.appointment_analytics.sql_call",
					args:{
						"from_date":report.filters[0].value,
						"to_date":report.filters[1].value,
						"args":x
					},
					callback:function(r){
						console.log(r.message)
						tableData = r.message
						console.log(tableData[0])
						const tableHtml = `
						<table class="table table-bordered table-striped">
							<thead >
								<tr>
									<th> Client name </th>
									<th>customer name</th>
									<th> status</th>
									<th> paid </th>
									<th> partially paid </th>
									<!-- Add more column headers as needed -->
								</tr>
							</thead>
							<tbody>
							${tableData
								.map(data => `<tr>
									<td><a href="http://2care.wajihah.sa/app/client-appointment-ct/${data.name}">${data.name}</a></td>
									<td>${data.customer}</td>
									<td>${data.status}</td>
									<td>${data['Billed:Int:100']}</td>
									<td>${data['partially Billed:Int:100']}</td>
								</tr>`)
								.join('')}
							</tbody>
						</table>
					`;
		
	
						let d = new frappe.ui.Dialog({
						title:"sub report",
						fields: [
							{
								fieldtype: "HTML",
								label: __("Table"),
								fieldname: "table",
							}
						]
	
						})
						d.fields_dict.table.$wrapper.html(tableHtml);
						d.show()
				
					}
				});
				
			});
	
	
	
			//console.log(report)
		},

    formatter: function(value, row, column, data, default_formatter) {
        value = default_formatter(value, row, column, data);
        
        // Check if this is the status column (lowercase fieldname from query)
        if (column.fieldname == "status" && value) {
            let status = value.toLowerCase().trim();
            let color = "gray";

            if (status == "attended" || status == "waiting attend") {
                color = "#17a2b8"; // Teal/cyan color like in Client Appointment CT
            }
            else if (status == "waiting" || status.includes("انتظار")) {
                color = "orange"; // Orange for waiting status
            }
            else if (status == "scheduled") {
                color = "green";
            }
            else if (status == "cancelled") {
                color = "red";
            }
            else if (status == "لم يحضر") {
                color = "gray";
            }

            value = `<span style="
                background:${color};
                color:white;
                padding:2px 8px;
                border-radius:12px;
                font-size:12px;
                display:inline-block;
            ">${value}</span>`;
        }

        return value;
    }
};